commitment.go

```diff
@@ -0,0 +1,100 @@
+package raft
+
+import (
+	"sort"
+	"sync"
+)
+
+// Commitment is used to advance the leader's commit index. The leader and
+// replication goroutines report in newly written entries with Match(), and
+// this notifies on commitCh when the commit index has advanced.
+type commitment struct {
+	// protectes matchIndexes and commitIndex
+	sync.Mutex
```

Comment:
Reading through it'll be minor, doesn't look like a caller can spend much time in a critical section